ExampleMod/Content/Buffs/ExampleLifeRegenDebuff.cs

```diff
@@ -0,0 +1,50 @@
+using Terraria;
+using Terraria.ModLoader;
+
+namespace ExampleMod.Content.Buffs
+{
+	// Fire debuff is an example of a buff that causes constant loss of life
+	// See ExampleLifeRegenDebuffPlayer.UpdateBadLifeRegen at the end of the file for more information
+	public class ExampleLifeRegenDebuff : ModBuff
+	{
+		public override void SetDefaults() {
+			DisplayName.SetDefault("Fire debuff"); // Buff display name
+			Description.SetDefault("Losing life"); // Buff description
+			Main.debuff[Type] = true;  // Is it a debuff?
+			Main.pvpBuff[Type] = true; // Players can give other players buffs, which are listed as pvpBuff
+			Main.buffNoSave[Type] = true; // It means the buff won't save when you exit the world
+			LongerExpertDebuff = true; // If this buff is a debuff, setting this to true will make this buff last twice as long on players in expert mode
+		}
+
+		// Allows you to make this buff give certain effects to the given player
+		public override void Update(Player player, ref int buffIndex) {
+			player.GetModPlayer<ExampleLifeRegenDebuffPlayer>().FireDebuff = true;
+		}
+	}
+
+	class ExampleLifeRegenDebuffPlayer: ModPlayer
+	{
+		// Flag checking when Fire Debuff should be activated
+		public bool FireDebuff;
+
+		public override void ResetEffects() {
+			FireDebuff = false;
+		}
+
+		// Allows you to give the player a negative life regeneration based on its state(for example, the "On Fire!" debuff makes the player take damage-over-time)
+		// This is typically done by setting player.lifeRegen to 0 if it is positive, setting player.lifeRegenTime to 0, and subtracting a number from player.lifeRegen
+		// The player will take damage at a rate of half the number you subtract per second
+		public override void UpdateBadLifeRegen() {
+			if (FireDebuff) {
```

Comment:
Though I myself may be starting to lean towards Microsoft's 'make everything public be in `UpperCamelCase`', the current tML style guide is asking fields to be in `lowerCamelCase`.
```suggestion
			player.GetModPlayer<ExampleLifeRegenDebuffPlayer>().lifeRegenDebuff= true;
		}
	}

	internal class ExampleLifeRegenDebuffPlayer : ModPlayer
	{
		// Flag checking when life regen debuff should be activated
		public bool lifeRegenDebuff;

		public override void ResetEffects() {
			fireDebuff = false;
		}

		// Allows you to give the player a negative life regeneration based on its state (for example, the "On Fire!" debuff makes the player take damage-over-time)
		// This is typically done by setting player.lifeRegen to 0 if it is positive, setting player.lifeRegenTime to 0, and subtracting a number from player.lifeRegen
		// The player will take damage at a rate of half the number you subtract per second
		public override void UpdateBadLifeRegen() {
			if (lifeRegenDebuff) {
```